app/presenters/admin/export/document_presenter.rb

```diff
@@ -0,0 +1,44 @@
+class Admin::Export::DocumentPresenter < Whitehall::Decorators::Decorator
+  include GovspeakHelper
+
+  def as_json(_options = {})
+    {
+      document: model,
+      editions: editions
+    }
+  end
+
+  private
+
+  def editions
+    model.editions.map do |edition|
+      edition_associations(edition)
+    end
+  end
+
+  def edition_associations(edition)
+    output = {
+               "edition": edition
+             }
+    output[:associations] = {}
+    associations = edition.class.reflect_on_all_associations.map(&:name)
+    associations.each do |association|
+      output[:associations][association] = edition.public_send(association)
+    end
+    output[:whitehall_admin_links] = resolve_whitehall_admin_links(edition.body)
+    output
+  end
+
+  def resolve_whitehall_admin_links(body)
+    whitehall_admin_links(body).map { |link|
```

Comment:
As this is a multi-line block it'd be more conventional to use do notation: https://rubystyle.guide/#single-line-blocks